docs/releases/4.2.txt

```diff
@@ -257,7 +257,22 @@ Templates
 Tests
 ~~~~~
 
-* ...
+* The classes :class:`~django.test.RequestFactory`,
+  :class:`~django.test.Client`, ``django.test.AsyncRequestFactory`` and
+  ``django.test.AsyncClient`` now support the ``headers`` parameter, which
+  allows to pass a dictionary with plain header names.
+  For :class:`~django.test.RequestFactory` and :class:`~django.test.Client`
+  classes, the header names can be specified without the ``HTTP_`` prefix.
+
+  .. code-block:: python
+
+     # Before:
+     Client().get("/path", HTTP_ACCEPT_LANGUAGE="fr")
+     AsyncClient().get("/path", ACCEPT_LANGUAGE="fr")
+
+     # After:
+     Client().get("/path", headers={"Accept-Language": "fr"})
+     AsyncClient().get("/path", headers={"Accept-Language": "fr"})
```

Comment:
1. Django URL’s typically end with a trailing slash, which the `APPEND_SLASH` feature helps with, we should try to follow that in examples
2. Best to use realistic data in examples, so use `/home/` instead of `/path`
3. Maybe better to show typical use of the test client with `self.client` / `self.async_client` - it's rarely necessary to construct a separate client.

```suggestion
     # Before:
     self.client.get("/home/", HTTP_ACCEPT_LANGUAGE="fr")
     await self.async_client.get("/home/', ACCEPT_LANGUAGE="fr")

     # After:
     self.client.get("/home/", headers={"Accept-Language": "fr"})
     await self.async_client.get("/home/", headers={"Accept-Language": "fr"})
```